core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/InternalDateRange.java

```diff
@@ -109,8 +117,8 @@ public Bucket createBucket(String key, double from, double to, long docCount, In
 
         @Override
         public Bucket createBucket(InternalAggregations aggregations, Bucket prototype) {
-            return new Bucket(prototype.getKey(), ((Number) prototype.getFrom()).doubleValue(), ((Number) prototype.getTo()).doubleValue(),
-                    prototype.getDocCount(), aggregations, prototype.getKeyed(), prototype.getFormat());
+            return new Bucket(prototype.getKey(), prototype.internalGetFrom(), prototype.internalGetTo(),
```

Comment:
We can not access the member directly because they are protected in InternalDateRange.Bucket's superclass and so InternalDateRange.Factory does not have access to it (well, at least not with jdk 1.8.0_91).